helix-core/src/main/java/org/apache/helix/monitoring/mbeans/ResourceMonitor.java

```diff
@@ -71,8 +71,9 @@
   private SimpleDynamicMetric<Long> _successTopStateHandoffCounter;
   private SimpleDynamicMetric<Long> _failedTopStateHandoffCounter;
   private SimpleDynamicMetric<Long> _maxSinglePartitionTopStateHandoffDuration;
-  private SimpleDynamicMetric<Long> _totalMessageReceived;
-
+  @Deprecated
+  private SimpleDynamicMetric<Long> _totalMessageReceived; // This should be counter but it is not now
```

Comment:
Nit: I think it would be good to add here why this needs to be a counter. We could add something like:

Because this is a metric indicating a "total" count (a.k.a. ever-increasing) value, we want to make sure this metric is a counter. If a gauge is used, the metric might get reset upon a controller switch, which won't be ever-increasing.

What do you think?

Reply:
Sure. Let me add the comment.